cachi2/core/package_managers/bundler/main.py

```diff
@@ -99,3 +106,36 @@ def _get_repo_name_from_origin_remote(package_dir: RootedPath) -> str:
 
     resolved_path = Path(repo_name).joinpath(package_dir.subpath_from_root)
     return str(resolved_path)
+
+
+def _prepare_environment_variables_for_hermetic_build() -> list[EnvironmentVariable]:
+    return [
+        # Contains path to a directory where a new config could be found.
+        EnvironmentVariable(name="BUNDLE_APP_CONFIG", value="${output_dir}/" + CONFIG_OVERRIDE),
+    ]
+
+
+def _prepare_for_hermetic_build(source_dir: RootedPath, output_dir: RootedPath) -> None:
+    """Prepare a package for hermetic build by injecting necessary config."""
```

Comment:
This prepare can't dump the file, because in order to do so, it would need to know the real target `${output_dir}` which we won't know until either `generate-env` or `inject-files` is run. Therefore, you'll need to split the logic and dump an intermediary data to the build config and only dump the file to disk with one of the other commands (most likely `inject-files`).

Reply:
Oh, it will be a different output_dir then, not the one I thought!